boa/src/syntax/parser/expression/mod.rs

```diff
@@ -28,6 +47,23 @@ use crate::syntax::ast::{node::Node, punc::Punctuator, token::TokenKind};
 ///
 /// Those exressions are divided by the punctuators passed as the third parameter.
 macro_rules! expression { ($name:ident, $lower:ident, [$( $op:path ),*], [$( $low_param:ident ),*] ) => {
+    impl $name {
+        // Helper method to build a BinOp in the parse method
+        fn build_op<O, L>(&self, op: O, lhs: L, cursor: &mut Cursor<'_>) -> ParseResult
+        where
+            O: TryInto<BinOp>,
+            <O as TryInto<BinOp>>::Error: Debug, // for call to `expect`
+            L: Into<Box<Node>>
+        {
+            let _ = cursor.next().expect("token disappeared");
```

Comment:
Moving the cursor in this function seems a bit strange. This function should expect that the cursor is ready to parse. It' s easier to reason that way. The `cursor.next()` if needed, should be done before calling it.